src/generate/formats/writer/yaml/utils.test.ts

```diff
@@ -407,3 +411,64 @@ describe("writeLoopString", () => {
     },
   );
 });
+
+const TEST_YAML_PATH = "src/commands/experimental/yaml/fixtures/specs/test-integration.yaml";
+
+describe("extractComponentList", () => {
+  it("extracts the right list of components", async () => {
+    const basePath = process.env.PWD ?? process.cwd();
+    process.chdir(basePath);
+    console.log("BASE PATH", basePath);
+    const integration = load(
+      await fs.readFile(TEST_YAML_PATH, "utf-8"),
+    ) as IntegrationObjectFromYAML;
+    const result = await extractComponentList(integration);
+    expect(result).toMatchObject({
+      branch: {
+        isPublic: true,
+        registryPrefix: "@component-manifests",
+        version: "LATEST",
+      },
+      code: {
+        isPublic: true,
+        registryPrefix: "@component-manifests",
+        version: "LATEST",
+      },
+      log: {
+        isPublic: true,
+        registryPrefix: "@component-manifests",
+        version: "LATEST",
+      },
+      loop: {
+        isPublic: true,
+        registryPrefix: "@component-manifests",
+        version: "LATEST",
+      },
```

Comment:
Thinking back on our convo and the existing shuffling of responsibilities - should `extractComponentList` also be the thing that says "CNI doesn't care about this component" instead of having it in the two downstream callsites?